web/views.py

```
@@ -981,6 +1180,178 @@ def course_progress_overview(request, slug):
     return render(request, "courses/progress_overview.html", context)
 
 
+@login_required
+def progress_visualization(request):
```

Ruff (0.8.2)

Redefinition of unused `progress_visualization` from line 979

(F811)